lit_stablelm/model.py

```diff
@@ -153,17 +145,10 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         q = torch.cat((q_roped, q[..., n_elem:]), dim=-1)
         k = torch.cat((k_roped, k[..., n_elem:]), dim=-1)
 
-        if hasattr(self, "bias"):
-            # causal self-attention; Self-attend: (B, nh, T, hs) x (B, nh, hs, T) -> (B, nh, T, T)
-            # NOTE: cannot use flash attention because it takes q.size(-1) as the norm factor which is different to the
-            # head size when rotary_percentage is set
-            att = (q @ k.transpose(-2, -1)) * (1.0 / math.sqrt(head_size))
-            att = att.masked_fill(self.bias[:, :, :T, :T] == 0, float("-inf"))
-            att = F.softmax(att, dim=-1)
-            y = att @ v  # (B, nh, T, T) x (B, nh, T, hs) -> (B, nh, T, hs)
-        else:
-            # efficient attention using Flash Attention CUDA kernels
-            y = F.scaled_dot_product_attention(q, k, v, attn_mask=None, dropout_p=0.0, is_causal=True)
+        # efficient attention using Flash Attention CUDA kernels
+        y = F.scaled_dot_product_attention(
+            q, k, v, attn_mask=None, dropout_p=0.0, is_causal=True, scale=1.0 / math.sqrt(head_size)
```

Comment:
One of the CUDA tests is failing: `pytest tests/test_model.py::test_bfloat16_llama_init`

```python
E       RuntimeError: Expected query, key, and value to have the same dtype, but got query.dtype: float key.dtype: float and value.dtype: c10::BFloat16 instead.
```

I think there's a bug in how the dtype is managed in rope

Reply:
Opened #13